[bitnami/spark] Update to /bitnami/spark/README.md "Installing additional Jars" example

JamesMaki opened the pull request:
Updating the /bitnami/spark/README.md "Installing additional jars" example to work correctly.

### Description of the change

Curl is not included in the existing Bitnami Spark image, this updates the example to use a multi-stage Docker build to accomplish adding the additonal Jar file. 

### Benefits

The example in the documentation will now build and run correctly.

### Applicable issues
#54826

JamesMaki committed:
Update README.md

Updating the README.md "Installing additional jars" example to work correctly.

Signed-off-by: JMaki <7774872+JMaki3@users.noreply.github.com>

fevisera commented on bitnami/spark/README.md:
````diff
@@ -198,12 +198,15 @@ After that, your changes will be taken into account in the server's behaviour.
+By default, this container bundles a generic set of jar files but the default image can be extended to add as many jars as needed for your specific use case. For instance, the following Dockerfile adds [`aws-java-sdk-bundle-1.11.704.jar`](https://mvnrepository.com/artifact/com.amazonaws/aws-java-sdk-bundle/1.11.704) using a multi-stage Docker build:
 
 ```Dockerfile
+FROM curlimages/curl-base:latest as curl
````
There is a straightforward example for adding `curl` at https://github.com/bitnami/containers/issues/13905#issuecomment-1317724282:

```
FROM bitnami/spark
USER root
RUN install_packages curl
USER 1001
RUN curl https://repo1.maven.org/maven2/com/amazonaws/aws-java-sdk-bundle/1.11.704/aws-java-sdk-bundle-1.11.704.jar --output /opt/bitnami/spark/jars/aws-java-sdk-bundle-1.11.704.jar
```

In that case, a multi-stage build would not be necessary. Does it make sense to you to replace your example with this one?

fevisera commented on bitnami/spark/README.md:
```diff
@@ -198,12 +198,15 @@ After that, your changes will be taken into account in the server's behaviour.
 ### Installing additional jars
 
-By default, this container bundles a generic set of jar files but the default image can be extended to add as many jars as needed for your specific use case. For instance, the following Dockerfile adds [`aws-java-sdk-bundle-1.11.704.jar`](https://mvnrepository.com/artifact/com.amazonaws/aws-java-sdk-bundle/1.11.704):
+By default, this container bundles a generic set of jar files but the default image can be extended to add as many jars as needed for your specific use case. For instance, the following Dockerfile adds [`aws-java-sdk-bundle-1.11.704.jar`](https://mvnrepository.com/artifact/com.amazonaws/aws-java-sdk-bundle/1.11.704) using a multi-stage Docker build:
```
```suggestion
By default, this container bundles a generic set of jar files but the default image can be extended to add as many jars as needed for your specific use case. For instance, the following Dockerfile adds [`aws-java-sdk-bundle-1.11.704.jar`](https://mvnrepository.com/artifact/com.amazonaws/aws-java-sdk-bundle/1.11.704):
```

Because of the other comment, multi-stage would not be necessary.

fevisera commented:
Hi @JMaki3,

Kindly review the comments added to this PR.

Thank you!

JMaki3 commented:
Thanks @fevisera I don't have strong feelings either way, I am happy to update this example to use the `RUN install_packages curl` method instead if that is what you prefer. I just want the docs to be a bit easier for the next person who comes along.

JamesMaki committed:
Update bitnami/spark/README.md

Co-authored-by: Felipe V.C. Serafim <43246350+fevisera@users.noreply.github.com>
Signed-off-by: JMaki <7774872+JMaki3@users.noreply.github.com>

JamesMaki committed:
Update README.md

Updating example to add curl using install_packages instead of a multi-stage Docker build.

Signed-off-by: JMaki <7774872+JMaki3@users.noreply.github.com>

JamesMaki committed:
Update README.md

Updating a second example in the documentation that also needed to install curl.

Signed-off-by: JMaki <7774872+JMaki3@users.noreply.github.com>

JamesMaki reviewed (commented):
I also updated the example below this one that downloads and extracts a new version of Spark, that Dockerfile still does not build though. I think the link is inactive now.

fevisera commented on bitnami/spark/README.md:
````diff
@@ -214,7 +216,8 @@ Go to <https://spark.apache.org/downloads.html> and copy the download url bundli
 
 ```Dockerfile
````
Please update the Dockerfile with the following instructions which contain the link updated:

```Dockerfile
FROM bitnami/spark:3.5.0
USER root
RUN install_packages curl
USER 1001
RUN rm -r /opt/bitnami/spark/jars && \
    curl --location https://dlcdn.apache.org/spark/spark-3.5.0/spark-3.5.0-bin-hadoop3.tgz | \
    tar --extract --gzip --strip=1 --directory /opt/bitnami/spark/ spark-3.5.0-bin-hadoop3/jars/
```

fevisera commented:
Hi @JMaki3,

Kindly review the comment. The link is the one provided from https://spark.apache.org/downloads.html.

Thank you.

JamesMaki committed:
Update README.md

Updating Hadoop jars example

Signed-off-by: JMaki <7774872+JMaki3@users.noreply.github.com>

JMaki3 commented:
@fevisera I made the change you recommended, thank you.

fevisera reviewed (approved):
Thank you for applying the changes.
LGTM!
